src/instrumentation.rs

```diff
@@ -0,0 +1,108 @@
+//! APIs wrapping the Python interpreter's instrumentation features.
+use crate::ffi;
+use crate::pyclass::boolean_struct::False;
+use crate::types::PyFrame;
+use crate::{Bound, PyAny, PyClass, PyObject, PyRefMut, PyResult, Python};
+use std::ffi::c_int;
+
+/// Represents a monitoring event used by the profiling API
+pub enum ProfileEvent<'py> {
+    /// A python function or method was called or a generator was entered.
+    Call,
+    /// A python function or method returned or a generator yielded. The
+    /// contained data is the value returned to the caller or `None` if
+    /// caused by an exception.
+    Return(Option<Bound<'py, PyAny>>),
+    /// A C function is about to be called. The contained data is the
+    /// function object being called.
+    CCall(Bound<'py, PyAny>),
```

Comment:
Ooh, maybe!

Reply:
It seems a little more correct to do this so I think we might as well, but I'm not sure it really unlocks any useful functionality, since `PyCFunction` is a pretty opaque type.